src/server.jl

```diff
@@ -42,10 +42,11 @@ end
 
 struct Server
     workers::Dict{String,File}
-
+    lock::ReentrantLock # should be locked for mutation/lookup of the workers dict, not for evaling on the workers. use worker lock for that
+    workerlocks::Dict{String,ReentrantLock}
```

Comment:
What's the reason the worker lock cannot live directly in the `File` objects?

Reply:
I did that because you need to lock access to a file before it exists. But I'm thinking now that you could in principle pass the pre-locked lock to the `File` constructor as well if we wanted to store it in there.

Reply:
It seems to me better to maintain the locks internally to each file, rather than having to maintain two different dicts.

Reply:
I had it like that first, and changed it during some refactoring. But I didn't try it again after arriving at the current solution. I'll give it another shot.

Reply:
hm the problem is that I don't want to do the file initialization within the server lock, because it can potentially lock it for quite a bit of time. But I'd have to do that, because otherwise I can't signal that I'm locking a not-yet-existing `File`. Unless I add a placeholder `PlaceholderFile` or so that can be placed in the workers dict. But that also seems a bit complicated.

Reply:
Ok as the file initialization takes a second or so and we are not talking about a high-load server, I put file initialization under the server lock after all, which at least simplifies the double-dict structure.